salt/_modules/metalk8s_kubernetes.py

```diff
@@ -226,6 +226,9 @@ def method(
         try:
             result = method_func(**call_kwargs)
         except (ApiException, HTTPError) as exc:
+            # Handle 409 conflict error for replace action, send it back to the state
+            if exc.status == 409:
+                return "409"
```

Comment:
From what I see HTTPError does not have a `status` field

I would prefer to have this logic in `handle_error` directly and check that the exception is `ApiException`

And also I do not really like the simple return of a string, I would prefer to raise a classic salt exception and catch it based on the message in the state, something like that (not tested)

``` python
# in the module
raise CommandExecutionError("409 Conflict") from exception

# in the state
try:
    new = <classic replace call ...>
except CommandExecutionError as exc:
    if exc.message == "409 Conflict":
        # LET'S RETRY !!
        continue
    raise
```